streams/src/main/java/org/apache/kafka/streams/internals/metrics/StreamsClientMetricsDelegatingReporter.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.internals.metrics;
+
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.common.metrics.KafkaMetric;
+import org.apache.kafka.common.metrics.MetricsReporter;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+public class StreamsClientMetricsDelegatingReporter implements MetricsReporter {
+
+    private static final Logger log = LoggerFactory.getLogger(StreamsClientMetricsDelegatingReporter.class);
+    private final Admin adminClient;
+
+    public StreamsClientMetricsDelegatingReporter(final Admin adminClient) {
+        this.adminClient = Objects.requireNonNull(adminClient);
+        log.debug("Creating Client Metrics reporter for admin client {}", adminClient);
```

Comment:
What does `adminClient.toString()` return -- anything useful? What information do we intent to log?

For the StreamThread case we log
```
 log.debug("Creating MetricsReporter for threadId {} and stateUpdaterId {}", threadId, stateUpdaterThreadId);
 ```

 So maybe log `streamsClientId` instead of `admin` so we align the code? Should be easy to pass in a new parameter similar to the StreamThread case?

Reply:
Yep, that sounds good.  For `streamsClientId` you mean the instance id assigned to the admin client for telemetry?

Reply:
Yes.